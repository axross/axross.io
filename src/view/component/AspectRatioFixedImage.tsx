import { ClassAttributes, createElement } from 'react';
import styled from '../../core/emotion';

type AspectRatio = [number, number];

type Props = {
  src: string;
  aspectRatio: AspectRatio;
  width?: string;
  height?: string;
  className?: string;
};

const AspectRatioFixedImage = ({
  src,
  aspectRatio,
  width = '100%',
  height = 'auto',
  className,
}: ClassAttributes<HTMLElement> & Props) => (
  <Root _width={width} _height={height} className={className}>
    <Inner src={src} aspectRatio={aspectRatio} />
  </Root>
);

const Root = styled<{ _width: string; _height: string }, 'div'>('div')`
  width: ${({ _width }) => _width};
  height: ${({ _height }) => _height};
`;

const Inner = styled<{ src: string; aspectRatio: AspectRatio }, 'div'>('div')`
  padding-top: ${({ aspectRatio }) => 100 / aspectRatio[0] * aspectRatio[1]}%;
  background-image: url(${({ src }) => src});
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center center;
`;

export default AspectRatioFixedImage;
