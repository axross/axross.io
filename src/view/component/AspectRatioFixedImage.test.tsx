import { render } from 'enzyme';
import { createElement } from 'react';
import AspectRatioFixedImage from './AspectRatioFixedImage';

describe('<AspectRatioFixedImage>', () => {
  test('it matches with the previous snapshot', () => {
    const rendered = render(<AspectRatioFixedImage src="https://loremflickr.com/320/240" aspectRatio={[320, 240]} />);

    expect(rendered).toMatchSnapshot();
  });
});
