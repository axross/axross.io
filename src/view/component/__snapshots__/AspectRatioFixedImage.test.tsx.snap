// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AspectRatioFixedImage> it matches with the previous snapshot 1`] = `
.emotion-1 {
  width: 100%;
  height: auto;
}

.emotion-0 {
  padding-top: 75%;
  background-image: url(https://loremflickr.com/320/240);
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center center;
}

<div
  class="emotion-1"
>
  <div
    class="emotion-0"
    src="https://loremflickr.com/320/240"
  />
</div>
`;
